llvm/include/llvm/Support/GenericDomTree.h

```diff
@@ -831,9 +827,10 @@ class DominatorTreeBase {
   void addRoot(NodeT *BB) { this->Roots.push_back(BB); }
 
   DomTreeNodeBase<NodeT> *createChild(NodeT *BB, DomTreeNodeBase<NodeT> *IDom) {
-    return (DomTreeNodes[BB] = IDom->addChild(
-                std::make_unique<DomTreeNodeBase<NodeT>>(BB, IDom)))
-        .get();
+    auto Node = std::make_unique<DomTreeNodeBase<NodeT>>(BB, IDom);
+    auto *NodePtr = (DomTreeNodes[BB] = std::move(Node)).get();
+    IDom->addChild(NodePtr);
+    return NodePtr;
```

Comment:
Can we do something like this?
```suggestion
    auto *NodePtr = Node.get();
    DomTreeNodes[BB] = std::move(Node);
    return NodePtr;
```
Really not a fan of this "using return value of move assignment operator" pattern...

Reply:
+1

Reply:
Changed. (I didn't like that either, but wanted at least some consistency with createNode below.) I now merged this method with createNode to avoid code duplication.